code/modules/power/solar.dm

```diff
@@ -293,73 +306,103 @@
 		return TRUE
 
-	if(istype(W, /obj/item/stack/sheet/glass) || istype(W, /obj/item/stack/sheet/rglass))
-		if(!anchored)
-			to_chat(user, span_warning("You need to secure the assembly before you can add glass."))
-			return
-		var/turf/solarturf = get_turf(src)
-		if(locate(/obj/machinery/power/solar) in solarturf)
-			to_chat(user, span_warning("A solar panel is already assembled here."))
+	if(tracker)
+		if(item_used.tool_behaviour == TOOL_CROWBAR)
+			new /obj/item/electronics/tracker(src.loc)
+			tracker = FALSE
+			update_appearance()
+			user.visible_message(span_notice("[user] takes out the electronics from the solar assembly."), span_notice("You take out the electronics from the solar assembly."))
+			return TRUE
+
+		if(!istype(item_used, /obj/item/stack/sheet/glass))
+			to_chat(user, span_warning("The tracker only accepts standard, un-reinforced glass."))
 			return
-		var/obj/item/stack/sheet/S = W
-		if(S.use(2))
-			glass_type = W.type
-			playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
-			user.visible_message(span_notice("[user] places the glass on the solar assembly."), span_notice("You place the glass on the solar assembly."))
-			if(tracker)
-				new /obj/machinery/power/tracker(get_turf(src), src)
-			else
-				new /obj/machinery/power/solar(get_turf(src), src)
-		else
-			to_chat(user, span_warning("You need two sheets of glass to put them into a solar panel!"))
+		var/obj/item/stack/sheet/my_sheet = item_used
+		if(!my_sheet.use(2))
+			to_chat(user, span_warning("You don't have enough glass to complete the tracker."))
 			return
-		return TRUE
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
+		user.visible_message(span_notice("[user] places the glass on the solar assembly."),span_notice("You place the glass on the solar assembly."))
+		new /obj/machinery/power/tracker/(get_turf(src))
+		return
 
 	if(!tracker)
-		if(istype(W, /obj/item/electronics/tracker))
-			if(!user.temporarilyRemoveItemFromInventory(W))
+		if(istype(item_used, /obj/item/electronics/tracker))
+			if(!user.temporarilyRemoveItemFromInventory(item_used))
 				return
 			tracker = TRUE
 			update_appearance()
-			qdel(W)
+			qdel(item_used)
 			user.visible_message(span_notice("[user] inserts the electronics into the solar assembly."), span_notice("You insert the electronics into the solar assembly."))
 			return TRUE
-	else
-		if(W.tool_behaviour == TOOL_CROWBAR)
-			new /obj/item/electronics/tracker(src.loc)
-			tracker = FALSE
-			update_appearance()
-			user.visible_message(span_notice("[user] takes out the electronics from the solar assembly."), span_notice("You take out the electronics from the solar assembly."))
-			return TRUE
+
+	//make a list of all the glass
+	var/list/acceptable_glass_list
+	acceptable_glass_list = typecacheof(list(
+		/obj/item/stack/sheet/glass,
+		/obj/item/stack/sheet/plasmaglass,
+		/obj/item/stack/sheet/titaniumglass,
+		/obj/item/stack/sheet/plastitaniumglass,
+	))
```

Comment:
```suggestion
	var/static/list/acceptable_glass_list = typecacheof(list(
		/obj/item/stack/sheet/glass,
		/obj/item/stack/sheet/plasmaglass,
		/obj/item/stack/sheet/titaniumglass,
		/obj/item/stack/sheet/plastitaniumglass,
	))
```
No need to split that over two lines. You should also make it static, so typecacheof isn't called every time this code is ran (and typecacheof always gives the same result)